geopandas/geodataframe.py

```diff
@@ -1422,6 +1408,35 @@ def apply(self, func, axis=0, raw=False, result_type=None, args=(), **kwargs):
 
     @property
     def _constructor(self):
+        geometry_default = getattr(self, "_geometry_column_name")
+        crs_default = getattr(self, "crs")
+
+        def _geodataframe_constructor_with_fallback(
+            data=None, index=None, crs=crs_default, geometry=geometry_default, **kwargs
+        ):
+            df = pd.DataFrame(data, index=index)
```

Comment:
This is potentially wasteful, as it's not necessarily used - if we have a geometry column, we reinitialise df as a GeoDataFrame below. But we need a way to get all columns with key `geometry` and this seemed less flakey than handling dict, dataframe and blockmanager directly.

Reply:
The way it was before (first creating GeoDataFrame, and then converting back to DataFrame when needed) would avoid this? (but then of course the case where we fall back to DataFrame has an additional conversion step)